### PR TITLE
Fix bld2repo

### LINKED ISSUE #42: bld2repo should download subpackages needed for build
`python2-six-1.11.0-6.module+el8.4.0+9287+299307c7` needs `python2-libs-2.7.18-2.module+el8.4.0+9193+f3daf6ef.x86_64.rpm` for build but the following command does not download it:
```
bld2repo --build-id 1444241 \
    --koji-host "https://brewhub.engineering.redhat.com/brewhub" \
    --koji-storage-host http://download.devel.redhat.com/brewroot"
```

... which prevents the package from being successfully built in mock:
```
mock -r rhel-8-x86_64 --quiet --installdeps" "/tmp/csmockiodxf3_u/python2-six-1.11.0-6.module+el8.4.0+9287+299307c7.src.rpm" "--addrepo" "file:///tmp/csmockiodxf3_u/local-build-repo"
Error: 
 Problem: package python2-tkinter-2.7.18-2.module+el8.4.0+9193+f3daf6ef.x86_64 requires python2 = 2.7.18-2.module+el8.4.0+9193+f3daf6ef, but none of the providers can be installed
  - conflicting requests
  - nothing provides python2-libs(x86-64) = 2.7.18-2.module+el8.4.0+9193+f3daf6ef needed by python2-2.7.18-2.module+el8.4.0+9193+f3daf6ef.x86_64
(try to add '--skip-broken' to skip uninstallable packages
```

### PR DESCRIPTION
**Pull request by fivaldi**
This should fix #42 .

### CODE REVIEW

File: bld2repo/bld2repo/__init__.py [hunk 1]
```diff
@@ -45,7 +45,6 @@ def get_buildrequire_pkgs_from_build(build_id, session, config):
         for rpm in tagged_rpms:
             if pkg["build_id"] == rpm["build_id"] and rpm["arch"] in archs:
                 pkg_md["rpms"].append(rpm)
-                tagged_rpms.remove(rpm)
```
Comment (FrostyX) [inline bld2repo/bld2repo/__init__.py:48]: Not sure if it is a good idea to remove items from the currently iterated list.

Comment (fivaldi) [reply bld2repo/bld2repo/__init__.py:48]: Indeed, it's not a good idea. That's why I removed this line which removes items from the currently iterated list. :-)
During testing, the missing packages/RPMs reported by @kdudka appeared to be downloaded. So it seems to fix the issue.
obrazek

**Comment by FrostyX** [comment]
> That's why I removed this line which removes items from the currently iterated list. :-)

Ah, some temporal brain dysfunction on my side :D
(I though you are adding it)

> So this patch seems to fix the issue.

+1 then :-)

**Comment by fivaldi** [comment]
No problem & thanks for merging.

**Pull request merged**